tests/test-maximize.lua

```diff
@@ -142,6 +153,90 @@ local steps = {
         assert(geo_to_str(original_geo) == geo_to_str(new_geo),
             geo_to_str(original_geo) .. " == " .. geo_to_str(new_geo))
 
+        c:kill()
+
+        return true
+    end,
+    -- Now, start some clients maximized
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_before=true})
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if not c then return end
+
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical)
+        assert(c.maximized)
+
+        c:kill()
+
+        return true
+    end,
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_after=true})
+
+        return true
+    end,
+    function()
+        local c = client.get()[1]
+
+        if not c then return end
+
+        assert(not c.maximized_horizontal)
+        assert(not c.maximized_vertical)
+
+        -- There is a small 1 frame delay to lower the risk of a race condition
+        if c.maximized then
+            c:kill()
+            return true
+        end
+    end,
+    function()
+        if #client.get() > 0 then return end
+
+        test_client(nil,nil,nil,nil,nil,{maximize_after=true})
```

Comment:
Why are you testing `maximize_after` twice? This and the following function does the same thing as the previous two function (modulo something that seems to just be preparing for the following test).